dataflowengine/src/main/scala/io/shiftleft/dataflowengine/language/nodemethods/TrackingPointMethods.scala

```diff
@@ -0,0 +1,31 @@
+package io.shiftleft.dataflowengine.language.nodemethods
+
+import io.shiftleft.codepropertygraph.generated.nodes
+import io.shiftleft.dataflowengine.language.TrackingPoint
+import io.shiftleft.semanticcpg.language._
+import io.shiftleft.dataflowengine.language._
+import io.shiftleft.semanticcpg.language.nodemethods.generalizations.TrackingPointToCfgNode
+
+class TrackingPointMethods(val node: nodes.TrackingPointBase) extends AnyVal {
+
+  /**
+    * Convert to nearest CFG node for flow pretty printing
+    * */
+  def cfgNode: nodes.CfgNode = {
+    node.accept(TrackingPointToCfgNode)
+  }
+
+  /**
+    * Convert to nearest AST node
+    * */
+  def astNode: nodes.AstNode = {
+    node match {
+      case n: nodes.AstNode               => n
+      case n: nodes.DetachedTrackingPoint => n.cfgNode
```

Comment:
you're aware that this will throw a MatchError for all other cases? compiler might be issuing a warning as well

Reply:
Yes, I am aware. This covers all cases, and I didn't introduce a default case so that the compiler warns if this is ever no longer the case.